src/loki/status_action.cc

```diff
@@ -0,0 +1,16 @@
+#include "loki/worker.h"
+
+namespace valhalla {
+namespace loki {
+void loki_worker_t::status(Api&) const {
+#ifdef HAVE_HTTP
+  // if we are in the process of shutting down we signal that here
+  // should react by draining traffic (though they are likely doing this as they are usually the ones
+  // who sent us the request to shutdown)
+  if (prime_server::draining() || prime_server::shutting_down()) {
+    throw valhalla_exception_t{102};
```

Comment:
if any process detects that its in the process of shutting down we throw this error which turns into an HTTP 503 with a message about the server shutting down